prow/config/config.go

```diff
@@ -382,10 +382,16 @@ func (c *Config) mergeJobConfig(jc JobConfig) error {
 		validPeriodics[p.Name] = true
 	}
 
-	// validate no presubmit with same name exists cross multiple files
-	validPresubmits := map[string]bool{}
+	// validate presubmit with same name do not run on the same branches across multiple files
+	validPresubmits := map[string]Presubmit{}
 	for _, p := range c.AllPresubmits(nil) {
-		validPresubmits[p.Name] = true
+		if existingJog, ok := validPresubmits[p.Name]; ok {
+			fmt.Println(existingJog.Brancher)
+			if existingJog.Brancher.Intersects(p.Brancher) {
+				return fmt.Errorf("duplicated presubmit from main config file : %s", p.Name)
```

Comment:
this check will become expensive (n^2) when we have more and more small config files... We should only do once for main config, and assume merged config is good.

Reply:
If you are concerned about performance, you might as well merge all the file and check at the end.